scripts/ImportExportTest.sh

```diff
@@ -0,0 +1,276 @@
+
+    if [ -n "${_TESTED}" ]
+    then
+        TESTED=$((TESTED + 1))
+    fi
+}
+
+# function tests whether exporting and importing again leaves the JSON ast unchanged
+# Results are recorded by adding to FAILED or UNCOMPILABLE.
+# Also, in case of a mismatch a diff and the respective ASTs are printed
+# Expected parameters:
+# $1 name of the file to be exported and imported
+# $2 any files needed to do so that might be in parent directories
+function testImportExportEquivalence {
+    local sol_file="$1"
+    local input_files="$2"
+    if $SOLC --bin "${input_files}" > /dev/null 2>&1
+    then
+        ! [[ -e stderr.txt ]] || { fail "stderr.txt already exists. Refusing to overwrite."; }
+
+        if [[ $IMPORT_TEST_TYPE == "ast" ]]
+        then
+            Ast_ImportExportEquivalence "${sol_file}" "${input_files}"
+        elif [[ $IMPORT_TEST_TYPE == "evm-assembly" ]]
+        then
+            JsonEvmAsm_ImportExportEquivalence "${sol_file}" "${input_files}"
+        else
+            fail "Unknown import test type. Aborting."
+        fi
+    else
+        UNCOMPILABLE=$((UNCOMPILABLE + 1))
+    fi
+}
+
+WORKINGDIR=$PWD
+NSOURCES=0
+
+command_available "${SOLC}" --version
+command_available jq --version
+
+# for solfile in $(find $DEV_DIR -name *.sol)
+# boost_filesystem_bug specifically tests a local fix for a boost::filesystem
+# bug. Since the test involves a malformed path, there is no point in running
+# AST tests on it. See https://github.com/boostorg/filesystem/issues/176
+if [[ $IMPORT_TEST_TYPE == "ast" ]]
+then
+    TEST_DIRS=("${SYNTAXTESTS_DIR}" "${ASTJSONTESTS_DIR}")
+elif [[ $IMPORT_TEST_TYPE == "evm-assembly" ]]
+then
+    TEST_DIRS=("${SYNTAXTESTS_DIR}" "${SEMANTICTESTS_DIR}")
+else
+    fail "Unknown import test type. Aborting. Please specify ${0} [ast|evm-assembly]."
+fi
+
+IMPORT_TEST_FILES=$(find "${TEST_DIRS[@]}" -name "*.sol" -and -not -name "boost_filesystem_bug.sol")
+
+NSOURCES="$(echo "$IMPORT_TEST_FILES" | wc -l)"
+echo "Looking at $NSOURCES .sol files..."
+
+for solfile in ${IMPORT_TEST_FILES}
+do
+    echo -n "."
+    # create a temporary sub-directory
+    FILETMP=$(mktemp -d)
+    cd "$FILETMP"
+
+    set +e
+    OUTPUT=$("$SPLITSOURCES" "$solfile")
+    SPLITSOURCES_RC=$?
+    set -e
+    if [ ${SPLITSOURCES_RC} == 0 ]
+    then
+        OIFS=${IFS}
+        IFS=' ' read -ra OUTPUT_ARRAY <<< "${OUTPUT}"
+        IFS=${OIFS}
+        NSOURCES=$((NSOURCES - 1 + ${#OUTPUT_ARRAY[@]}))
+        testImportExportEquivalence "$solfile" "${OUTPUT[@]}"
+    elif [ ${SPLITSOURCES_RC} == 1 ]
+    then
+        testImportExportEquivalence "$solfile" "$solfile"
+    elif [ ${SPLITSOURCES_RC} == 2 ]
+    then
+        # The script will exit with return code 2, if an UnicodeDecodeError occurred.
+        # This is the case if e.g. some tests are using invalid utf-8 sequences. We will ignore
+        # these errors, but print the actual output of the script.
+        echo >&2
+        printError "\n${OUTPUT[*]}\n"
```

Comment:
`$OUTPUT` is not an array. Or did you mean `${OUTPUT_ARRAY[*]}` here? I'd use `${OUTPUT}` anyway.
```suggestion
        printError "\n${OUTPUT}\n"
```
Same below.